SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift

```diff
@@ -0,0 +1,196 @@
+    
+    private func setupEvents() {
+        headerCollectionView.delegate = self
+        headerCollectionView.dataSource = self
+        bodyCollectionView.delegate = self
+        bodyCollectionView.dataSource = self
+        
+        viewModel.$indexPath
+            .sink { _ in
+                print("something went wrong in scrollCollectionView")
+            } receiveValue: { [weak self] indexPath in
+                self?.headerCollectionView.scrollToItem(at: indexPath, at: .left, animated: false)
+                self?.bodyCollectionView.scrollToItem(at: indexPath, at: .left, animated: false)
+            }
+            .store(in: &cancellables)
+    }
+    
+    private func setupLayout() {
+        // header
+        headerCollectionView.translatesAutoresizingMaskIntoConstraints = false
+        headerCollectionView.isUserInteractionEnabled = false
+        view.addSubview(headerCollectionView)
+        (headerCollectionView.collectionViewLayout as? UICollectionViewFlowLayout)?.scrollDirection = .horizontal
+        headerCollectionView.register(HeaderCollectionViewCell.self, forCellWithReuseIdentifier: "HeaderCollectionViewCell")
+        
+        NSLayoutConstraint.activate([
+            headerCollectionView.topAnchor.constraint(equalTo: view.topAnchor),
+            headerCollectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            headerCollectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            headerCollectionView.heightAnchor.constraint(equalToConstant: 50)
+        ])
+        
+        // body
+        bodyCollectionView.translatesAutoresizingMaskIntoConstraints = false
+        bodyCollectionView.isPagingEnabled = true
+        bodyCollectionView.backgroundColor = .clear
+        view.addSubview(bodyCollectionView)
+        (bodyCollectionView.collectionViewLayout as? UICollectionViewFlowLayout)?.scrollDirection = .horizontal
+        bodyCollectionView.register(BodyCollectionViewCell.self, forCellWithReuseIdentifier: "BodyCollectionViewCell")
+        
+        NSLayoutConstraint.activate([
+            bodyCollectionView.topAnchor.constraint(equalTo: headerCollectionView.topAnchor),
+            bodyCollectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            bodyCollectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            bodyCollectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+}
+
+extension HomeViewController: UICollectionViewDelegateFlowLayout {
+    ///
+    /// The layout of Header & Body.
+    ///
+    /// Header
+    ///         []  []  []
+    /// Body
+    ///         [][][][][]
+    ///
+    /// When the Body swips `5` blocks, the Header needs to swipe `2` blocks.
+    ///
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard scrollView == bodyCollectionView else { return }
+        headerCollectionView.contentOffset.x = scrollView.contentOffset.x * (2 / 5)
```

Comment:
The numbers 5 and 2 appear to have specific significance in this context and are being consistently used. 
I recommend declaring them as constants for better clarity.

Reply:
Agree, I'll update it.

Reply:
[done](https://github.com/No-57/SCSH_3C_iOS/pull/27/commits/da17810c430e4915f0b6ae0008e3b491fdc2b755)